src/app/zap-templates/templates/app/CHIPClientCallbacks.zapt

```diff
@@ -11,11 +11,11 @@
 #include <lib/support/Span.h>
 
 // List specific responses
-{{#all_user_clusters side='client'}}
+{{#zcl_clusters}}
```

Comment:
So for "normal" applications (like say a light-switch) this is now going to try to generate all possible callbacks in the header, even through the application only uses a few of them, right?

If this is not going to depend on the _app_ config, then it should not be under "app" and should instead move to app-common.

Reply:
This generates `CHIPClientCallbacks.h` and the only usage of that I have seen was in JNI code, so I assumed 
this is JNI-specific.

I agree that placement seems off, I noticed the issue only because java started to fail compilation.

I will look into moving it. It is unfortunate that we grab this from data_model (which is app-specific).

Reply:
It's generated for all applications; you just can't see for the others because they are compile-time generated.

Now it's possible none of those other applications _use_ this file (because they are mostly not doing any interesting client stuff), which would be a very good reason to either stop generating it or move it to some jni-specific codegen bit.